core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -295,9 +335,16 @@ export class OptimizedSsrEngine {
     });
   }
 
-  protected log(message: string, debug = true): void {
-    if (!debug || this.ssrOptions?.debug) {
-      console.log(message);
+  protected log(
+    message: string,
+    debug = true,
+    logMetadata?: ExpressServerLoggerContext
```

Comment:
[MINOR]

let's add a TODO note (not to miss it in the future), because `... || ({} as ExpressServerLoggerContext)` in line 346 is just a temporary backward-compatible workaround

```suggestion
    //CXSPA-3680 - in a new major, let's make this argument required
    logMetadata?: ExpressServerLoggerContext
```